flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  description = "A flake for s2n-tls";
+
+  inputs.nixpkgs.url = github:NixOS/nixpkgs/nixos-22.11;
+
+  outputs = { self, nix, nixpkgs, flake-utils }: 
+     flake-utils.lib.eachDefaultSystem (system:
+      let pkgs = nixpkgs.legacyPackages.${system};
+   in rec {
+      packages.s2n-tls = pkgs.stdenv.mkDerivation {
+        src = self;
+        name = "s2n-tls";
+        inherit system; 
+        # TODO: override the s2n-tls version
+
+        buildInputs = [ pkgs.cmake
```

Comment:
Not a blocker for this but we'll need to figure out how to override CC inputs as well. Ideally we are testing against the oldest GCC/clang versions that we support.

Reply:
Yes, up next, and believe it'll take the form of an override.  In  the clang/llvm case, it appears to require more than one nixpkg.